docs/enhancements/add-external-providers.md

```diff
@@ -0,0 +1,182 @@
+resources requirements.
+
+Assisted Installer is also ideal for mixed env configurations.
+** The RHV provider supports installing a mixed env where the Control-Plain (Master) nodes are deployed on top of RHV
+and Compute (Worker) nodes are Bare-Metal nodes.
+
+### Goals
+
+- Create a Provider interface, and hooks in the assisted installer code which separates the assisted installer core code
+ from the provider-specific code.
+- Add clear documentation for each provider specific function.
+- Document steps to install a provider cluster with assisted installer
+
+### Non-Goals
+
+- Support IPI installations.
+- Integrating all relevant [Openshift installer types](https://github.com/openshift/installer/tree/master/pkg/types)
+into the assisted installer code - this will require a large refactor of the assisted installer code.
+- Remove Baremetal from not appropriate stuct/functions names(example in open questions).
+ 
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a RHV/VSphere admin I want to create an Openshift cluster on my RHV engine by using the assisted installer.
+
+#### Story 2
+
+As a RHV admin I want to create an Openshift cluster on my RHV engine by using the assisted installer in which
+Control-Plain (Master) nodes are deployed on top of RHV and Compute (Worker) nodes are Bare-Metal nodes.
+
+#### Story 3
+
+As an Openshift developer working on platform X I want to add support for installing Openshift with assisted installer
+on top of platform X.
+
+### Implementation Details/Notes/Constraints
+
+#### Changes to the assisted-service project:
+
+##### Step 1: Detect platform specific logic
+
+Go over the assisted installer flow and detect parts of the code which require provider specific logic.
+
+##### Step 2: Create a Go interface
+
+We should create a Go interface that will contain each provider specific logic in a function
+for each provider to implement.
+
+For example, prototype:
+
+```go
+// Provider contains functions which are required to support installing on a specific platform.
+type Provider interface {
+	// Name returns the name of the platform.
+	Name() string
+	// UpdatePlatformInventoryData sets all platform specific inventory data on the updates map.
+	UpdatePlatformInventoryData(params installer.UpdateClusterParams, cluster *common.Cluster, updates map[string]interface{}) error
+    // SetPlatformInstallConfig sets all platform specific install config parameters
+    SetPlatformInstallcfgValues(platform *platformVsphere, clusterPlatform *models.VspherePlatform)
+    //...
+}
+```
+
+##### Step 3: Implement registry
+
+Implement the registry pattern so providers can register with their names.
+When we will detect the a certain provider then we will register it.
+If no provider is detected then we register the Bare-Metal provider.
+
+##### Step 4: Implement providerDetector method
+
+Implement a function which determines which provider should be used.
```

Comment:
```suggestion
Implement a function which determines which providers can be used.
```

Reply:
Regarding the can/should I think that can be the provider's decision on how to implement, but it is an interesting question to ask.
If there is a use case in which the user will provision the Nodes on a certain provider but want to disable provider integration? I'll add that to the open questions

Reply:
Sure.  We see that today.  Maybe they don't have platform credentials, or they are doing some custom integrations.